clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -472,14 +475,63 @@ static Value *EmitISOVolatileStore(CodeGenFunction &CGF, const CallExpr *E) {
   return Store;
 }
 
+static StringRef
+convertFPAccuracy(LangOptions::FPAccuracyKind FPAccuracy) {
+  StringRef AccuracyVal;
+  switch (FPAccuracy) {
+  case LangOptions::FPA_Default:
+    AccuracyVal = "fpaccuracy.default";
+    break;
+  case LangOptions::FPA_High:
+    AccuracyVal = "fpaccuracy.high";
+    break;
+  case LangOptions::FPA_Medium:
+    AccuracyVal = "fpaccuracy.medium";
+    break;
+  case LangOptions::FPA_Low:
+    AccuracyVal = "fpaccuracy.low";
+    break;
+  case LangOptions::FPA_Sycl:
+    AccuracyVal = "fpaccuracy.sycl";
+    break;
+  case LangOptions::FPA_Cuda:
+    AccuracyVal = "fpaccuracy.cuda";
+    break;
+  }
+  return AccuracyVal;
+}
+
 // Emit a simple mangled intrinsic that has 1 argument and a return type
 // matching the argument type. Depending on mode, this may be a constrained
 // floating-point intrinsic.
 static Value *emitUnaryMaybeConstrainedFPBuiltin(CodeGenFunction &CGF,
                                 const CallExpr *E, unsigned IntrinsicID,
-                                unsigned ConstrainedIntrinsicID) {
+                                unsigned ConstrainedIntrinsicID,
+    unsigned FPAccuracyIntrinsicID = Intrinsic::not_intrinsic) {
+  llvm::TargetLibraryInfoImpl TLII(CGF.getTarget().getTriple());
   llvm::Value *Src0 = CGF.EmitScalarExpr(E->getArg(0));
-
+  if (FPAccuracyIntrinsicID != Intrinsic::not_intrinsic) {
+    LangOptions::FPAccuracyKind FPAccuracy = CGF.getLangOpts().getFPAccuracy();
+    // Enter this part of the condition only when TLI.isFPACCuracyAvailable is
+    // true, implying FPAccuary != LangOptions::FPA_Default.
```

Comment:
I don't think this assumption is correct. It's possible that we'll have a back-end that supports multiple accuracy implementations but we want to use the default accuracy anyway. In fact, that's probably going to be the most common case.